### PR TITLE
ER-378 - Recruit Vacancy Client

### PR DESCRIPTION
**Pull request by chrisjensenuk**
.NET version is the same as FAA (4.5.1)
Mongo driver is the same as that used by FAA (1.10.0)

`Client.cs` has a single public function `GetVacancy`. This returns a vacancy as it is in the query store.

`VacancyVersionHelper.cs` is a helper class to determine what version a vacancy is by its number. 10 digits means it is a v2 vacancy.

### CODE REVIEW

File: DAS.Recruit.Vacancies.Client/DAS.Recruit.Vacancies.Client/Client.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        private readonly string _connectionString;
+        private readonly string _databaseName;
+        private readonly string _collectionName;
+
+        static Client()
+        {
+            var pack = new ConventionPack
+            {
+                new CamelCaseElementNameConvention(),
+                new EnumRepresentationConvention(MongoDB.Bson.BsonType.String),
```
Comment (sachpatel) [inline DAS.Recruit.Vacancies.Client/DAS.Recruit.Vacancies.Client/Client.cs]: If we are persisting and reading back as a string, do we need this Enum convention.

File: DAS.Recruit.Vacancies.Client/DAS.Recruit.Vacancies.Client/VacancyVersionHelper.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DAS.Recruit.Vacancies.Client
+{
+    public static class VacancyVersionHelper
+    {
+        private const int V2VacancyIdLength = 10;
+
+        public static bool IsRecruitVacancy(long vacancyId)
```
Comment (sachpatel) [inline DAS.Recruit.Vacancies.Client/DAS.Recruit.Vacancies.Client/VacancyVersionHelper.cs]: Should the parameters on these methods be called `vacancyReference`.

File: DAS.Recruit.Vacancies.Client/DAS.Recruit.Vacancies.Client/Client.cs (hunk 1 above)
Comment (chrisjensenuk) [reply DAS.Recruit.Vacancies.Client/DAS.Recruit.Vacancies.Client/Client.cs]: We don't have Enums as yet but this config mirrors the config of Employer which felt right.

File: DAS.Recruit.Vacancies.Client/DAS.Recruit.Vacancies.Client/VacancyVersionHelper.cs (hunk 2 above)
Comment (chrisjensenuk) [reply DAS.Recruit.Vacancies.Client/DAS.Recruit.Vacancies.Client/VacancyVersionHelper.cs]: This is the 'id' from the pov of the legacy application. So from FAA a negative value will be passed in as the identifier. E.g. -1234 for v1 vacancies and a 10 digit number for v2 vacancies. So its not necessarily the VRN.

**Pull request merged**